plugins/bunch_o_snapshots/plugin.js

```diff
@@ -0,0 +1,416 @@
+		};
+    Settings.openDialog();
+    Settings.dialog.close(0);
+
+    Settings.addCategory(
+      "BoSetings",
+      (data = {
+        name: pluginData.title,
+      })
+    );
+    if (BosSettings == null) {
+      doLog("create settings");
+      BosSettings = {
+        BoSdevMode: new Setting("BoSdevMode", {
+          name: "BoSdevMode",
+          description:
+            "Developer mode, disables the actualy usage of the plugin, testing purposes only, DO NOT USE.",
+          category: "BoSetings",
+          value: currentSettings.devMode,
+          type: "toggle",
+          onChange(value) {
+            updateCurrentSettings();
+          },
+        }),
+        BoSoutputDirectory: new Setting("BoSoutputDirectory", {
+          name: "BoSoutputDirectory",
+          description: "Where all your screenshots will be saved.",
+          category: "BoSetings",
+          value: currentSettings.outputDirectory,
+          type: "text",
+          onChange(value) {
+            updateCurrentSettings();
+          },
+        }),
+        BoSoutputDirectorySelection: new Setting(
+          "BoSoutputDirectorySelection",
+          {
+            name: "Change directory",
+            description:
+              "Open a directory selection window to change your BoSoutputDirectory",
+            category: "BoSetings",
+            value: false,
+            type: "click",
+            click: function () {
+              let dialog = new Dialog({
+                title: "WIP",
+                id: "dirSelectionWIP",
+                width: img.naturalWidth + 50,
+                lines: [
+                  "This is a work in progress...<br>Can't change the directory like this yet, you'll have to write the directory manually.",
+                ],
+                buttons: ["dialog.close"],
+                onButton(result_index) {
+                  dialog.delete();
+                  return true;
+                },
+              });
+              dialog.show();
+              setTimeout(() => {
+                dialog.close(0);
+                dialog.delete();
+              }, 10000);
+            },
+          }
+        ),
+        BoSsaveMode: new Setting("BoSsaveMode", {
+          name: "BoSsaveMode",
+          description:
+            "Set the screenshot saving mode, more info about modes in Plugin's documentation.",
+          category: "BoSetings",
+          value: currentSettings.saveMode,
+          type: "select",
+          options: {
+            direct: "direct",
+            justSave: "justSave",
+            fullConfirmation: "fullConfirmation",
+          },
+          onChange(value) {
+            updateCurrentSettings();
+          },
+        }),
+      };
+    }
+    updateCurrentSettings();
+    Settings.openDialog();
+    Settings.dialog.close(0);
+  }
+
+  // When updating, apply meta data changes to plugins.json entry!
+  BBPlugin.register("plugin", {
```

Comment:
Please ensure ID consistency. The plugin directory currently has a different ID compared to the one specified in plugins.json, and the name here plus the file name is just "plugin" when it should be the proper plugin ID.